### PR TITLE
OCPBUGSM-24964 Change sufficient masters validation to include insufficient, pending-for-input an known statuses

### PR DESCRIPTION
**Pull request by ori-amizur**
/cc @avishayt 
/cc @gamli75

### CODE REVIEW

File: internal/cluster/validator.go [hunk 1]
```diff
@@ -307,16 +307,17 @@ func (v *clusterValidator) sufficientMastersCount(c *clusterPreprocessContext) v
 		minMastersNeededForInstallation = common.AllowedNumberOfMasterHostsInNoneHaMode
 	}
 
-	knownHosts, ok := MapHostsByStatus(c.cluster)[models.HostStatusKnown]
-	if !ok { //if no known hosts exist, there is no sufficient master count
-		return boolValue(false)
+	hosts := make([]*models.Host, 0)
+	for k, v := range MapHostsByStatus(c.cluster) {
+		if k != models.HostStatusDisabled {
```
Comment (gamli75) [inline internal/cluster/validator.go:312]: I think that we should ignore also disconnected 
If we have 3 masters and 1 is in a disconnected state - it's not enough

Comment (ori-amizur) [reply internal/cluster/validator.go:312]: The cluster will be in insufficient state anyway, so it doesn't make much difference.

Comment (avishayt) [reply internal/cluster/validator.go:312]: I don't agree, we have other checks for disconnected.  Here we are just counting hosts regardless of state.

**Review by avishayt — APPROVED** [review]

**Pull request merged**